src/services/item/service.ts

```diff
@@ -745,4 +747,29 @@ export class ItemService {
     );
     return { items: results.map(({ item }) => item), copies: results.map(({ copy }) => copy) };
   }
+
+  async reorder(
+    actor: Member,
+    repositories: Repositories,
+    itemId: string,
+    body: { previousItemId?: string },
+  ) {
+    const item = await this.get(actor, repositories, itemId);
+
+    return repositories.itemRepository.reorder(item, body.previousItemId);
+  }
+
+  /**
+   * Rescale order of children (of itemId's parent) if necessary
+   * @param member
+   * @param repositories
+   * @param itemId item whose parent get its children order rescaled if necessary
+   */
+  async rescaleOrder(member: Member, repositories: Repositories, item: Item) {
+    const parentId = getParentFromPath(item.path);
+    if (parentId) {
+      const parentItem = await this.get(member, repositories, parentId);
+      await repositories.itemRepository.rescaleOrder(parentItem);
```

Comment:
What would happen if the parent is deleted between theses 2 lines.
Using a SQL transaction should be a precondition to call this function.